lib/oban/plugins/reindexer.ex

```diff
@@ -93,6 +110,24 @@ defmodule Oban.Plugins.Reindexer do
     {:noreply, schedule_reindex(state)}
   end
 
+  # Validation
+
+  defp validate_schedule(schedule) do
+    Expression.parse!(schedule)
+
+    :ok
+  rescue
+    error in [ArgumentError] -> {:error, error}
+  end
+
+  defp validate_timezone(timezone) do
+    if is_binary(timezone) and match?({:ok, _}, DateTime.now(timezone)) do
+      :ok
+    else
+      {:error, "expected :timezone to be a known timezone, got: #{inspect(timezone)}"}
+    end
+  end
```

Comment:
I think this validator helper list will grow sooner than later, and some of them will be duplicated in the codebase; for example, the implementation of `validate_timezone` it's the same or very similar to the one found in `validate_opt({:timezone, timezone})`. So, what do you think about wrapping these `validate_*` helpers under a module like `Oban.ConfigValidator` or something similar? Again, this reminds me about `Ecto.Changeset` and their `validate_*` functions.

Reply:
Great point. I'll keep that in mind for a subsequent refactor, as I aim to expose an exception-free validation helper on Config as well.